src/searchbar.cpp

```diff
@@ -177,6 +199,17 @@ void SearchBarLineEdit::fetchMoreSuggestions()
 
 void SearchBarLineEdit::onScroll(int value)
 {
+    /* Scrolling using key_down past end will teleport scroller to the top.
+       We undo this here. Block signal to avoid recursion.
+    */
+    if (!m_completer.popup()->currentIndex().isValid())
+    {
+        auto old = m_completer.popup()->verticalScrollBar()->blockSignals(true);
+        m_completer.popup()->scrollToBottom();
+        m_completer.popup()->verticalScrollBar()->blockSignals(old);
+        return;
+    }
```

Comment:
The need for this is not clear. It almost defies the uniform handling of scrolling through suggestions with the keyboard.

Maybe you should install the event filter on `m_completer.popup()` instead of on (or in addition to) `SearchBarLineEdit`?

If you cannot find a way to eliminate this piece of code (but not at the expense of similar hack!) then please add a comment describing that there is something wrong (or something that we don't understand) happening here.

Reply:
I cannot pinpoint the source of what triggers the scrolling (been this way when I first started tackling the Issue), so unfortunately this might be here to stay. I honestly suspect it might be an internal private class at this point, and it is rather confusing...

Reply:
Done

Reply:
I deleted this piece of code and nothing seems to be broken. Maybe the bug is fixed by a later change in the PR (e.g. measures taken against flickering)?

Reply:
The code is more for the 'smoothness' of the fetching when using Key_Down. You might notice when the Key_Down triggers the fetching, the scroller jumps up then down, which creates a flicker. If you add this piece of code, it smoothly extends the suggestions. It is should be the same smoothness as the Key_PageDown.

You are correct that functionally it still achieves the end result.